utils/land-patterns.stanza

```diff
@@ -5,281 +5,406 @@ defpackage ocdb/land-patterns :
-  type = TH
-  shape = Circle(anchor, pad-r)
-  layer(Cutout()) = Circle(anchor, drill-r)
-  layer(SolderMask(Top)) = Circle(anchor, mask-r)
-  layer(SolderMask(Bottom)) = Circle(anchor, mask-r)
-
-public defn pth-pad (drill-r:Double, mask-r:Double) :
-  pth-pad(C, drill-r, mask-r, mask-r)
-
-public defn pth-pad (anchor:Anchor, drill-r:Double) :
-  pth-pad(anchor, drill-r, drill-r + 0.35, drill-r + 0.35)
-
-public defn pth-pad (drill-r:Double) :
-  pth-pad(C, drill-r)
+  ref-label(loc(0.0, 0.0))
+
+; Create a capsule with normalized rotation
+defn capsule (anchor:Anchor, width:Double, height:Double) -> Capsule:
+  if width >= height:
+    Capsule(anchor, width, height)
+  else:
+    Capsule(width, height, loc(0.0, 0.0, 90.0))
+
+; Create a rounded rectangle with normalized rotation
+defn round-rect (width:Double, height:Double, r:Double) -> RoundedRectangle:
+  if width >= height:
+    RoundedRectangle(width, height, r)
+  else: 
+    RoundedRectangle(height, width, r, loc(0.0, 0.0, 90.0))
+
+; Create a chamfered rectangle with normalized rotation
+defn chamfered (width:Double, height:Double, r:Double) -> ChamferedRectangle:
+  if width >= height:
+    ChamferedRectangle(width, height, r)
+  else:
+    ChamferedRectangle(height, width, r)
+
+;==============================================================================
+;=========================== Geometric Helpers ================================
+;==============================================================================
+; Create a rectangle out of lines
+public defn LineRectangle (w:Double, h:Double, xc:Double, yc:Double, line-w:Double) :
+  val w2 = w * 0.5
+  val h2 = h * 0.5
+  Line(line-w, [
+    Point(xc - w2, yc - h2)
+    Point(xc + w2, yc - h2)
+    Point(xc + w2, yc + h2)
+    Point(xc - w2, yc + h2)
+    Point(xc - w2, yc - h2)])
+
+; Create a rectangle with width, height, center x/y and default line width
+public defn LineRectangle (w:Double, h:Double, xc:Double, yc:Double) :
+  LineRectangle(w, h, xc, yc, MIN-SILKSCREEN-WIDTH)
+
+; Create a rectangle with width and height and default x/y position
+public defn LineRectangle (w:Double, h:Double) :
+  LineRectangle(w, h, 0.0, 0.0)
+
+; Create a rectangle from dims structure
+public defn LineRectangle (d:Dims): 
+  LineRectangle(x(d), y(d))
+
+; Create a rectangle around another shape
+public defn outline-bounding-box (s:Shape):
+  LineRectangle(dims(s))
+
+; Create a "+" symbol with a fixed length at a center point
+public defn PlusSymbol (center:Point, l:Double) -> Shape:
+  Union $ [
+    Line(MIN-SILKSCREEN-WIDTH, [center + Point(-0.5 * l, 0.0), center + Point(0.5 * l, 0.0)])
+    Line(MIN-SILKSCREEN-WIDTH, [center + Point(0.0, -0.5 * l), center + Point(0.0, 0.5 * l)])
+  ]
+
+; Create a 1mm x 1mm plus symbol at Point(x, y)
+public defn PlusSymbol (x:Double, y:Double) -> Shape:
+  PlusSymbol(Point(x, y), 1.0)
+
+; Create a row of poses
+public defn pose-row (origin:Pose, count:Int, pitch:Double) -> Seqable<Pose>:
```

Comment:
Please use the existing `row-locs` and `col-locs` utilities instead of redefining `pose-row` and `pose-column`.

Reply:
I'm still unclear how to use those functions and why they are not in ocdb. I need explicit control over the `x,y` positions of every pad starting from an origin point.